examples/mongocxx/mongodb.com/documentation_examples.cpp

```diff
@@ -187,6 +210,40 @@ static bool is_replica_set(const mongocxx::client& client) {
     return static_cast<bool>(reply.view()["setName"]);
 }
 
+static bool version_at_least(mongocxx::database& db,
+                             int minimum_major,
+                             int minimum_minor,
+                             int minimum_patch) {
+    using bsoncxx::builder::basic::kvp;
+    using bsoncxx::builder::basic::make_document;
+
+    auto resp = db.run_command(make_document(kvp("buildInfo", 1)));
+    auto version = resp.find("version")->get_string().value;
+    std::string major_string;
+    std::string minor_string;
+    std::string patch_string;
+    int split = 0;
+    for (auto i : version) {
+        if (i == '.') {
+            split += 1;
+            continue;
+        }
+        if (split == 0) {
+            major_string += i;
+        } else if (split == 1) {
+            minor_string += i;
+        } else if (split == 2) {
+            patch_string += i;
+        }
+    }
+    int server_major = std::stoi(major_string);
+    int server_minor = std::stoi(minor_string);
+    int server_patch = std::stoi(patch_string);
+
+    return server_major >= minimum_major && server_minor >= minimum_minor &&
```

Comment:
I believe this function has a bug in it. Suppose we are running server version 7.0.0, and we are setting a minimum version of 4.4.0. The return statement as written would do the following comparisons below. We can see that the return will be `false` because even though the major version is successfully checked to be greater, the minor version of the 7.0.0 version is 0, which is less than 4, and therefore the entire statement will return false.
```
return (7 >= 4) && (0 >= 4) && (0 >= 0);
```